src/main/java/org/mockito/internal/util/reflection/StaticFinalOverrider.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2007 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.internal.util.reflection;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+
+/**
+ * To set a value on a static final field, setting the field accessible
+ * by using {@link Field#setAccessible(boolean)} is not enough.
+ * Only if removing the field modifier {@link Modifier#STATIC}, a value can be set via reflection.
+ */
+public class StaticFinalOverrider {
+    private Field modifiersField = null;
+
+    public void enableWrite(Field field) {
+        if ((field.getModifiers() & (Modifier.FINAL + Modifier.STATIC)) == 0) {
+            return;
+        }
+
+        try {
+            modifiersField = Field.class.getDeclaredField("modifiers");
```

Comment:
I think this one should be initialized statically.

Reply:
TIL (same as above ;)): Foo.class.getDeclaredField("bar") returns a new Field instance for every call.
So I'm not sure if a static variable is The Right Thing (tm)

Reply:
So let's try with a local variable then.